scripts/o3de/o3de/manifest.py

```diff
@@ -604,6 +604,8 @@ def get_registered(engine_name: str = None,
 
     # check global first then this engine
     if isinstance(engine_name, str):
+        if engine_name in json_data['engines_path']:
+            return pathlib.Path(json_data['engines_path'][engine_name]).resolve()
```

Comment:
Actually can you put this logic after the for loop.

The `engines` array should be the definitive location where the issue is registered at with us using the `engines_path` dictionary as a shortcut.

Also this code needs to protect against the `engine_paths` key not being in the load manifest for some reason as well.

So this should use `json_data.get('engines_path', {})` instead to protect against any KeyErrors